### ISSUE TITLE
node: missing readline.createInterface API

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_std#1511]
```
$ export DENO_NODE_COMPAT_URL=file://dev/deno_std/
$ npm install eslint
$ deno run --compat --unstable --allow-read --allow-env ./node_modules/eslint/bin/eslint.js --init
...
Oops! Something went wrong! :(

ESLint: 8.0.1

TypeError: fn.emit is not a function
    at Enquirer.enquirer.emit (file:///Users/biwanczuk/dev/vite_test/node_modules/enquirer/index.js:215:12)
    at Enquirer.ask (file:///Users/biwanczuk/dev/vite_test/node_modules/enquirer/index.js:127:10)
    at Enquirer.prompt (file:///Users/biwanczuk/dev/vite_test/node_modules/enquirer/index.js:82:20)
    at Function.fn (file:///Users/biwanczuk/dev/vite_test/node_modules/enquirer/index.js:218:23)
    at promptUser (file:///Users/biwanczuk/dev/vite_test/node_modules/eslint/lib/init/config-initializer.js:472:21)
    at Object.initializeConfig (file:///Users/biwanczuk/dev/vite_test/node_modules/eslint/lib/init/config-initializer.js:705:16)
    at main (file:///Users/biwanczuk/dev/vite_test/node_modules/eslint/bin/eslint.js:126:57)
    at Object.<anonymous> (file:///Users/biwanczuk/dev/vite_test/node_modules/eslint/bin/eslint.js:135:2)
    at Module._compile (file:///Users/biwanczuk/dev/deno_std/node/module.ts:251:36)
    at Object.Module._extensions..js (file:///Users/biwanczuk/dev/deno_std/node/module.ts:1276:10)
```

This is function that errors:
```ts
  static get prompt() {
    const fn = (questions, ...rest) => {
      let enquirer = new this(...rest);
      let emit = enquirer.emit.bind(enquirer);
      enquirer.emit = (...args) => {
        fn.emit(...args); // error here
        return emit(...args);
      };
      return enquirer.prompt(questions);
    };
    utils.mixinEmitter(fn, new Events());
    return fn;
  }
```

Something is going wrong when `utils.mixinEmitter(fn, new Events());`.

In addition to that I'm pretty sure this will break on printing interactive prompts due to missing APIs in `readline` module (`node/readline.js`)

**Comment by uki00a** [comment]
In Node.js, classes such as `EventEmitter` are written in ES5 style. This allows methods to be enumerated by `Object.keys()`:

```js
Object.keys(EventEmitter.prototype);
// In Node.js, the following array is returned:
// [
//   '_events',             '_eventsCount',
//   '_maxListeners',       'setMaxListeners',
//   'getMaxListeners',     'emit',
//   'addListener',         'on',
//   'prependListener',     'once',
//   'prependOnceListener', 'removeListener',
//   'off',                 'removeAllListeners',
//   'listeners',           'rawListeners',
//   'listenerCount',       'eventNames'
// ]
```

In contrast, the `EventEmitter` class in `deno_std/node/events.ts` is written in ES6 style, so the methods are not enumerable. :cry:

As a quick workaround, how about the following?

```ts
function makeMethodsEnumerable(klass: { new(): unknown }): void {
  const proto = klass.prototype;
  for (const key of Reflect.ownKeys(proto)) {
    const value = proto[key];
    if (typeof value === "function") {
      const desc = Reflect.getOwnPropertyDescriptor(proto, key);
      if (desc) {
        desc.enumerable = true;
        Object.defineProperty(proto, key, desc);
      }
    }
  }
}

makeMethodsEnumerable(EventEmitter);
```

**Comment by bartlomieju** [comment]
@uki00a if your solution will make "eslint --init" work, or at least move us forward to another error then I'm strongly in favor of the solution

**Comment by uki00a** [comment]
The fix in #1530 causes another error:

```shell
$ DENO_NODE_COMPAT_URL=file:///home/uki00a/ghq/github.com/uki00a/deno_std/ deno run --compat --unstable --allow-read --allow-env ./node_modules/eslint/bin/eslint.js --init
Check file:///home/uki00a/ghq/github.com/uki00a/deno_std/node/global.ts
Check file:///home/uki00a/ghq/github.com/uki00a/deno_std/node/global.ts
Not implemented: process.on("uncaughtException")
Not implemented: process.on("unhandledRejection")
error: Uncaught (in promise) TypeError: readline.createInterface is not a function
  let rl = readline.createInterface({ terminal: true, input: stdin });
                    ^
    at Function.keypress.listen (file:///home/uki00a/ghq/github.com/uki00a/test-eslint-with-compat-mode/node_modules/enquirer/lib/keypress.js:202:21)
    at SelectPrompt.start (file:///home/uki00a/ghq/github.com/uki00a/test-eslint-with-compat-mode/node_modules/enquirer/lib/prompt.js:180:28)
    at SelectPrompt.initialize (file:///home/uki00a/ghq/github.com/uki00a/test-eslint-with-compat-mode/node_modules/enquirer/lib/prompt.js:220:16)
    at SelectPrompt.initialize (file:///home/uki00a/ghq/github.com/uki00a/test-eslint-with-compat-mode/node_modules/enquirer/lib/types/array.js:26:17)
    at async file:///home/uki00a/ghq/github.com/uki00a/test-eslint-with-compat-mode/node_modules/enquirer/lib/prompt.js:236:7
```

**Comment by bartlomieju** [comment]
@uki00a thanks for checking, this expected as I didn't port that API from `readline` module yet.
